src/main/java/com/nike/cerberus/service/KmsService.java

```diff
@@ -105,4 +111,46 @@ public String provisionKmsKey(final String iamRoleId,
 
         return result.getKeyMetadata().getArn();
     }
+
+    /**
+     * When a KMS key policy statement is created and an AWS ARN is specified as a principal,
+     * AWS behind the scene binds that ARNs ID to the statement and not the ARN.
+     *
+     * They do this so that when a role or user is deleted and another team or person recreates the identity
+     * that this new identity does not get permissions that were not meant for it.
+     *
+     * In Cerberus's case we need to support a flow where teams use automation to automatically destroy and recreate IAM roles.
+     *
+     * We can detect that this event has happened because when an Identity that was referenced by an ARN in a KMS policy
+     * statement has been deleted the ARN is replaced by the ID. We can validate that principal matches an ARN pattern
+     * or recreate the policy.
+     *
+     * @param keyId - The CMK Id to validate the policies on.
+     * @param iamRoleArn - The Role ARN that should have decrypt permission
+     * @param kmsCMKRegion - The region that the key was provisioned for
+     */
+    public void validatePolicy(String keyId, String iamRoleArn, String kmsCMKRegion) {
+        AWSKMSClient kmsClient = kmsClientFactory.getClient(kmsCMKRegion);
+        GetKeyPolicyResult policyResult = null;
+        try {
+            policyResult = kmsClient.getKeyPolicy(new GetKeyPolicyRequest().withKeyId(keyId).withPolicyName("default"));
+        } catch (AmazonServiceException e) {
+            throw ApiException.newBuilder()
+                    .withApiErrors(DefaultApiError.FAILED_TO_VALIDATE_KMS_KEY_POLICY)
+                    .withExceptionCause(e)
+                    .withExceptionMessage(
+                            String.format("Failed to validate KMS key policy for keyId: " +
+                                    "%s for IAM role: %s in region: %s", keyId, iamRoleArn, kmsCMKRegion))
+                    .build();
+        }
+
+        if (! kmsPolicyService.isPolicyValid(policyResult.getPolicy(), iamRoleArn)) {
+            String updatedPolicy = kmsPolicyService.generateStandardKmsPolicy(iamRoleArn);
```

Comment:
Maybe we should log info() here or above when we find invalid ones?  It isn't going to happen that much and might be helpful.